plugins/cargo/cargo.go

```diff
@@ -7,18 +7,16 @@ import (
 	"github.com/1Password/shell-plugins/sdk/schema/credname"
 )
 
-var commands = [][]string{
-	{"publish"},
-	{"yank"},
-	{"owner"},
-}
-
 func CargoCLI() schema.Executable {
 	return schema.Executable{
-		Name:      "Cargo CLI",
-		Runs:      []string{"cargo"},
-		DocsURL:   sdk.URL("https://doc.rust-lang.org/cargo/index.html"),
-		NeedsAuth: needsauth.ForCommands(commands...),
+		Name:    "Cargo CLI",
+		Runs:    []string{"cargo"},
+		DocsURL: sdk.URL("https://doc.rust-lang.org/cargo/index.html"),
+		NeedsAuth: needsauth.OnlyFor(
+			needsauth.ForCommand("publish"),
+			needsauth.ForCommand("yank"),
+			needsauth.ForCommand("owner"),
+		),
```

Comment:
There should be an exception for `--help` when using these commands.

Reply:
This would mean (`ForCommand(publish)` OR `ForCommand(yank)` OR `ForCommand(owner)`) AND `NotForHelp()` (as Horia [stated](https://github.com/1Password/shell-plugins/pull/126#issuecomment-1387166231) the `For` is a logical AND and `OnlyFor` is logical `OR`).

So we could do  
```
needsauth.For(
  needsauth.OnlyFor(
			  needsauth.ForCommand("publish"),
			  needsauth.ForCommand("yank"),
			  needsauth.ForCommand("owner"),
  ),
  needsauth.NotForHelp()
)
```
But I would say this is also a good argument for renaming these functions `OnlyFor` -> `IfEitherMatches `, `For`-> `IfAllMatch`